store/governance_proposals.store.ts

```diff
@@ -281,15 +331,25 @@ export const useGovernanceProposalsStore = defineStore({
         proposal.createdTimestamp = Number(block.timestamp);
         proposal.createdDatetimeFormatted = formatDate(new Date(Number(block.timestamp) * 1000));
         proposal.createdBlockNumber = event.blockNumber;
-
-        // const voteStartDate = new Date(Number(proposal.voteStart) * 1000);
-        // const voteEndDate = new Date(Number(proposal.voteEnd) * 1000);
-        // console.log("Vote Start Date: ", voteStartDate);
-        // console.log("Vote End Date: ", voteEndDate);
+        // keep track of the proposal executed timestamp and block number if the proposal is executed
+        const executedProposal = this.fundProposals[this.web3Store.chainId][this.fundStore.fund?.address][proposal.proposalId];
```

Comment:
this line will crash if for example this.fundStore.fund? is undefined, can you make it work if any of array index lookups [idx] is undefined or doesn't exist?

Reply:
solved!